### PR TITLE
Task 1152-1159: BARD analogical reasoning dataset

### PR DESCRIPTION
**Pull request by ashok-arjun**
Total of 8 tasks.

Dataset link: https://github.com/NancyFulda/BYU-Analogical-Reasoning-Dataset

### CODE REVIEW

**Comment by danyaljj** [comment]
@davidstap  wondering if you can help with reviewing this PR?

**Comment by davidstap** [comment]
LGTM! I think the task is interesting, and the `Positive Examples` and `Negative Examples` are all clear. 

I would suggest some small changes:
* I think the `Description` can be improved by making it a bit more verbose: for me it was unclear, and I needed the Examples to fully understand the task. 
An improved `Description` could for example look like:
"An analogy that relates actions with their consequences is given in the form A: B. Your task is to replace the question mark (?) with the appropriate consequence of an action, following the A: B relation."
(Instead of:
"An analogy that relates actions with their consequences is given. Your task is to replace the question mark (?) with the appropriate consequence of the given action")

* I also think it would be more clear if the `input` pattern looked like `"cut : bleed. crush : ?"` (instead of `"input": "cut : bleed. crush ?"`).

**Comment by ashok-arjun** [comment]
@davidstap Thanks for the review! I've made the required changes. @danyaljj

File: tasks/task1152_bard_analogical_reasoning_causation.json [hunk 1]
```diff
@@ -0,0 +1,1272 @@
+{
+    "Contributors": [
+        "Arjun Ashok"
+    ],
+    "Source": [
+        "https://github.com/NancyFulda/BYU-Analogical-Reasoning-Dataset"
+    ],
+    "Definition": "An analogy that relates actions with their consequences is given in the form A : B. Your task is to replace the question mark (?) with the appropriate consquence of the given action, following the A: B relation.",
```
Comment (danyaljj) [inline tasks/task1152_bard_analogical_reasoning_causation.json]: The mention of "?" comes suddenly and has no pretext. May say that you're giving two pairs of actions in the following form `A:B . C:?`. And then follow discuss what needs to be done with ?.

Comment (danyaljj) [reply tasks/task1152_bard_analogical_reasoning_causation.json]: You can use similar ideas for the rest of the definitions to make them self-contained.

Comment (ashok-arjun) [reply tasks/task1152_bard_analogical_reasoning_causation.json]: Done

File: tasks/task1152_bard_analogical_reasoning_causation.json [hunk 2]
```diff
@@ -0,0 +1,1272 @@
+{
+    "Contributors": [
+        "Arjun Ashok"
+    ],
+    "Source": [
+        "https://github.com/NancyFulda/BYU-Analogical-Reasoning-Dataset"
+    ],
+    "Definition": "Two analogies that relate actions with their consequences is given in the form A : B. C : ?. Your task is to replace the question mark (?) with the appropriate consquence of the given action, following the A: B relation.",
```
Comment (danyaljj) [inline tasks/task1152_bard_analogical_reasoning_causation.json]: Now you need to bring back a sentence that defined what ":" between "A:B" means.

Comment (ashok-arjun) [reply tasks/task1152_bard_analogical_reasoning_causation.json]: I don't understand, that sentence wasn't there before. Can you please point me to the sentence?

Comment (danyaljj) [reply tasks/task1152_bard_analogical_reasoning_causation.json]: https://github.com/allenai/natural-instructions-expansion/blob/5c04862ff727d7d1dbc81cbe304cfb37ceff79d0/tasks/task1152_bard_analogical_reasoning_causation.json#L8

Comment (ashok-arjun) [reply tasks/task1152_bard_analogical_reasoning_causation.json]: Done! https://github.com/allenai/natural-instructions-expansion/blob/c6b3683bd7513ed94a3040bd18a341b81b0d977e/tasks/task1152_bard_analogical_reasoning_causation.json#L8

File: tasks/task1155_bard_analogical_reasoning_trash_or_treasure.json [hunk 3]
```diff
@@ -0,0 +1,3324 @@
...
+        {
+            "input": "wrapper : trash. peel : ?",
+            "output": [
+                "trash"
+            ]
+        },
+        {
+            "input": "wrapper : trash. toy : ?",
+            "output": [
+                "treasure"
```
Comment (danyaljj) [inline tasks/task1155_bard_analogical_reasoning_trash_or_treasure.json:62]: ?

File: tasks/task1155_bard_analogical_reasoning_trash_or_treasure.json [hunk 4]
```diff
@@ -0,0 +1,3324 @@
+{
+    "Contributors": [
+        "Arjun Ashok"
+    ],
+    "Source": [
+        "https://github.com/NancyFulda/BYU-Analogical-Reasoning-Dataset"
+    ],
+    "Definition": "Two analogies that relate items to whether they are trash or treasure is given in the form \"A : B. C : ?\". \"A : \"B relates item A to whether it is trash or treasure, as specified by B. Your task is to replace the question mark (?) with either `trash` or `treasure` for the given item C, following the \"A : B\" relation.",
```
Comment (danyaljj) [inline tasks/task1155_bard_analogical_reasoning_trash_or_treasure.json]: not quite sure if I understand this task.

**Comment by danyaljj** [comment]
Thanks! 

Now, except the "trash" vs "treasure" task, everything else looks good.

**Comment by ashok-arjun** [comment]
The task is related to reasoning between different objects - if A is a treasure, and B is related to A in a **specific manner** (which the model must know), it will also be a treasure.

The analogy is provided for this purpose.

Eg.

mud = trash. ring = treasure.

i.e. If mud is trash, is ring trash or treasure?

To answer this, the model must know that mud and ring are related in one way, and if mud is trash, ring should be treasure. So I think this is a valuable task.

If this does not make sense, let me know - I'll remove the task @danyaljj

File: tasks/task1155_bard_analogical_reasoning_trash_or_treasure.json (hunk 4 above)
Comment (danyaljj) [reply tasks/task1155_bard_analogical_reasoning_trash_or_treasure.json]: You need to properly define the task, by defining what it means to be "trash" or "treasure": 

 - trash ->   should be disposed of
 - treasure -> it a precious item which must be retained

Comment (ashok-arjun) [reply tasks/task1155_bard_analogical_reasoning_trash_or_treasure.json]: @danyaljj Done.https://github.com/allenai/natural-instructions-expansion/blob/e96b49cda3db0ecc30bd35f3750786d1409caf0a/tasks/task1155_bard_analogical_reasoning_trash_or_treasure.json#L8

Can this be merged now?

**Comment by danyaljj** [comment]
Looks good, thanks!

**Pull request merged**